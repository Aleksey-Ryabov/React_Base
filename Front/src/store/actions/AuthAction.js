

export const AuthValid = (data)=> ({
    type: 'ACCESS_ALLOW',
    payload: data
})

export const AuthNotValid = (data)=> ({
    type: 'ACCESS_DENIED',
    payload: data
})


export const fetchAuth = (user) => async(dispatch) => {
    const response = await fetch('http://localhost:3001/auth', {
        method: 'POST',
        headers: {"Content-Type": "application/json;charset=utf-8",},
        body: JSON.stringify(user),
    });
    const data = await response.json();
    console.log(data, 'action response data');

    if(data == true) {
        dispatch(AuthValid(data))
    }else {
        dispatch(AuthNotValid(data))
    }
}