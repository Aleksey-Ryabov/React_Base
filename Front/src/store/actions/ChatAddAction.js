


export const ChatAddAction = ()=> ({
    type: 'ADD_CHAT',
    payload: {id : parseInt( Math.random()* 200)}
})

export const ChatAddComment = (id, user, comment)=> ({
    type: 'ADD_CHAT_COMMENT',
    payload : {id, user, comment}
})

export const ChatDelete = (id)=> (
    {
        type: 'REMOVE_CHAT',
        payload: id
    }
)

export default ChatAddAction;