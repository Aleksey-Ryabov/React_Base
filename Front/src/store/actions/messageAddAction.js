
export const addUser = (user)=> ({
    type: 'ADD_USER',
    payload: user
})

export const addComment = (comment)=> ({
    type: 'ADD_COMMENT',
    payload: comment
})
