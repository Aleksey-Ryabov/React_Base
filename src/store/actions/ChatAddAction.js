

export const ChatAddAction = (id)=> ({
    type: 'ADD_CHAT',
    payload:  {id: parseInt(Math.random()*100), type: 'ЧАТ № 1'}
})

export const ChatDelete = (id)=> (
    {
        type: 'REMOVE_CHAT',
        payload: id
    }
)

export default ChatAddAction;
