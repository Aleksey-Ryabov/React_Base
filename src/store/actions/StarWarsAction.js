

export const getHeroes = (heroes) => ({
    type: 'ADD_HERO',
    payload: heroes
})

export const getError = (error) => ({
    type: 'GET_ERROR',
    payload: {flag: true, text: error}
})

export const AddHero = () => (dispatch, getState) => {
    const randomHero =  parseInt(Math.random()* 200)
        fetch(`https://swapi.dev/api/people/${randomHero}/`)
        .then(response => {
            if(!response.ok){
                throw new Error(`Request failed with status ${response.status}`);
            }
            return response.json()
        })
        .then((elem) => dispatch(getHeroes(elem)))
        .catch((error)=> dispatch(getError(error.message)))
}
