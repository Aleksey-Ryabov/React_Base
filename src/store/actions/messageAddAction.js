

export const messageAddAction = (userComment,commentUser)=> ({
    type: 'ADD_MESSAGE',
    payload:  {
        id: parseInt(Math.random()*100),
        user: userComment,
        comment: commentUser,
    }
})

export default messageAddAction;