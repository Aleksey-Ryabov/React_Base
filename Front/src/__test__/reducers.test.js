import { createSlice } from '@reduxjs/toolkit'
import { addUser } from '../store/actions/messageAddAction';
import messageReduce from '../store/reducers/messages';


const initialState = [
    {
        user: '', 
        comment: ''
    }
  ]

it('returns correct state after ADD_USER action', () => {
    const userStore = messageReduce(initialState, addUser('new User'));
    expect(userStore).toMatchSnapshot();
  });

