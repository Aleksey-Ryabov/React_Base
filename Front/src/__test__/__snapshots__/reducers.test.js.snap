// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns correct state after ADD_USER action 1`] = `
Object {
  "0": Object {
    "comment": "",
    "user": "",
  },
  "user": "new User",
}
`;
